### PR TITLE
fix(active-campaign): add check before iterating stored event array

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

Added check before iterating stored event array for Active campaign

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/active_campaign/transform.js [hunk 1]
```diff
@@ -503,11 +503,13 @@ const trackRequestHandler = async (message, category, destination) => {
...
   }
 
-  const storedEventsArr = res.response.data.eventTrackingEvents;
+  const storedEventsArr = res.response?.data?.eventTrackingEvents;
   const storedEvents = [];
-  storedEventsArr.map((ev) => {
-    storedEvents.push(ev.name);
-  });
+  if (Array.isArray(storedEventsArr)) {
+    storedEventsArr.map((ev) => {
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:509]: Can we refactor to the below way ?
```javascript
const storedEvents = Array.isArray(storedEventsArr) ? storedEventsArr.map(ev => ev.name) : [];
```

**Review by krishna2020 — APPROVED** [review]

**Review by sanpj2292 — DISMISSED** [review]

File: src/v0/destinations/active_campaign/transform.js (hunk 1 above)
Comment (utsabc) [reply src/v0/destinations/active_campaign/transform.js:509]: yes

**Review by ItsSudip — APPROVED** [review]

File: src/v0/destinations/active_campaign/transform.js [hunk 2]
```diff
@@ -503,11 +503,13 @@ const trackRequestHandler = async (message, category, destination) => {
...
+      storedEvents.push(ev.name);
+    });
+  }
```
Comment (saikumarrs) [inline src/v0/destinations/active_campaign/transform.js:512]: The names `storedEventsArr` and `storedEvents` are confusing. Can you rename them for avoiding it?

**Review by saikumarrs — APPROVED** [review]

File: src/v0/destinations/active_campaign/transform.js [hunk 3]
```diff
@@ -426,13 +426,10 @@ const screenRequestHandler = async (message, category, destination) => {
   }
 
   const storedEventsArr = res.response.data.eventTrackingEvents;
```
Comment (saikumarrs) [inline src/v0/destinations/active_campaign/transform.js:428]: Implement optional chaining here as well.

Comment (ItsSudip) [reply src/v0/destinations/active_campaign/transform.js]: Done

**Review by saikumarrs — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**